src/ResourceManager/Advisor/Commands.Advisor/Cmdlets/DisableAzureRmAdvisorRecommendation.cs

```diff
@@ -0,0 +1,172 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceGraph.Cmdlets
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+    using Advisor.Cmdlets.Models;
+    using Advisor.Cmdlets.Utilities;
+    using Advisor.Utilities;
+    using Management.Advisor.Models;
+    using Rest.Azure;
+
+    /// <summary>
+    /// Disable-AzureRmAdvisorRecommendation cmdlet
+    /// </summary>
+    /// <seealso cref="Microsoft.Azure.Commands.ResourceGraph.Utilities.ResourceGraphBaseCmdlet" />
+    [Cmdlet("Disable", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AdvisorRecommendation", DefaultParameterSetName = IdParameterSet), OutputType(typeof(List<PsAzureAdvisorSuppressionContract>))]
+    public class DisableAzureRmAdvisorRecommendation : ResourceAdvisorBaseCmdlet
+    {
+        /// <summary>
+        /// Constant for IdParameterSet
+        /// </summary>
+        public const string IdParameterSet = "IdParameterSet";
+
+        /// <summary>
+        /// Constant for NameParameterSet
+        /// </summary>
+        public const string NameParameterSet = "NameParameterSet";
+
+        /// <summary>
+        /// Constant for InputObjectParameterSet
+        /// </summary>
+        public const string InputObjectParameterSet = "InputObjectParameterSet";
+
+        /// <summary>
+        /// Default suppression name for the suppression API
+        /// </summary>
+        public const string DefaultSuppressionName = "HardcodedSuppressionName";
+
+        /// <summary>
+        /// Gets or sets the Resource Id.
+        /// </summary>
+        [Parameter(ParameterSetName = "IdParameterSet", Mandatory = true, HelpMessage = "ResourceID of the recommendation to be suppressed (space delimitited).")]
+        [Alias("Id")]
+        public string ResourceId { get; set; }
```

Comment:
I did not add it in the first place because we will not be using the ResourceId in the piped scenario, We only use PS object data sent through pipeline.

I added it just incase, let me know if we want to remove it if there are no use-case.

Reply:
You need to support ResourceId piping: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/piping-in-powershell.md.  Now that you added the attribute, it looks like it is correctly enabled.